packages/storybook/src/components/validityState/validityState.ts

```diff
@@ -0,0 +1,93 @@
+import { type TemplateResult, html } from 'lit-html';
+
+type OdsComponent = {
+  getValidity: () => ValidityState,
+} & HTMLElement
+
+let oldIsRequired: boolean;
+const odsTable = html`
+<ods-table id="validity-state-table" size="sm">
+  <table style="width: 200px">
+    <thead>
+      <tr>
+        <th scope="col">Key</th>
+        <th scope="col">Value</th>
+      </tr>
+    </thead>
+    <tbody>
+    </tbody>
+  </table>
+</ods-table>`;
+
+function ValidityStateTemplateDemo(hasValidityState: boolean, isRequired: boolean, componentName: string, componentSelector: string): TemplateResult | string {
+  if (oldIsRequired !== isRequired) {
+    oldIsRequired = isRequired;
+    const event = new Event('updateValidityState');
+    document.dispatchEvent(event);
+  }
+  const validityStateTemplate = html`
+  ${ odsTable }
+  <script>
+    (async () => {
+        await Promise.all([
+          await customElements.whenDefined('ods-table'),
+          await customElements.whenDefined('ods-${componentName}'),
+        ]);
+        await window.renderValidityState('${componentSelector}');
+    })();
+  </script>`;
+  return hasValidityState ? validityStateTemplate : '';
+}
+
+function ValidityStateTemplateExample(componentName: string, componentSelector: string): TemplateResult | string {
+  const validityStateTemplate = html`
+  ${ odsTable }
+  <script>
+    (async () => {
+        await Promise.all([
+          await customElements.whenDefined('ods-table'),
+          await customElements.whenDefined('ods-${componentName}'),
+        ]);
+        await window.renderValidityState('${componentSelector}');
+    })();
+  </script>`;
+  return validityStateTemplate;
+}
+
+async function renderValidityState(componentSelector: string): Promise<void> {
+  const component = document.querySelector<OdsComponent>(componentSelector);
+  component?.addEventListener('odsChange', async() => {
+    console.log('odsChange', )
+    await buildTable(component);
+  });
+  document.addEventListener('updateValidityState', async() => {
+    await new Promise((resolve) => setTimeout(resolve, 20));
```

Comment:
20??

Reply:
With 0 it don't work because the change (is-required on the control) is end in storybook but not in the component